### PR TITLE
[ADVAPP-995]: Improve the UI/UX design of the student profile page

### PR DESCRIPTION
**Pull request by danharrin**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-995

### Technical Description

Rebuilds the student profile page with an improved design.

### Any deployment steps required?

No

### Are any Feature Flags and/or Data Migrations that can eventually be removed Added?

No

_______________________________________________

### CODE REVIEW

File: app-modules/student-data-model/resources/views/filament/resources/student-resource/view-student.blade.php [hunk 1]
```diff
@@ -31,39 +31,92 @@
...
+                        'interactions' => InteractionsRelationManager::class,
+                        'files' => EngagementFilesRelationManager::class,
+                    ]"
+                />
+
+                <div class="grid grid-cols-1 gap-8 xl:grid-cols-2">
+                    @if (EducatableAlertsWidget::canView())
+                        @livewire(EducatableAlertsWidget::class, [
+                            'educatable' => $this->getRecord(),
+                            'manageUrl' => \AdvisingApp\StudentDataModel\Filament\Resources\StudentResource::getUrl('alerts', ['record' => $this->getRecord()]),
```
[the remark this answers is not in the document]
Comment (danharrin) [reply app-modules/student-data-model/resources/views/filament/resources/student-resource/view-student.blade.php]: Done

File: app-modules/student-data-model/src/Filament/Resources/EducatableResource/Widgets/EducatableActivityFeedWidget.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+use Filament\Infolists\Concerns\InteractsWithInfolists;
+use AdvisingApp\Engagement\Enums\EngagementDeliveryMethod;
+use AdvisingApp\StudentDataModel\Models\Contracts\Educatable;
+use AdvisingApp\Timeline\Livewire\Concerns\HasTimelineRecords;
+use AdvisingApp\Timeline\Livewire\Concerns\CanLoadTimelineRecords;
+
+class EducatableActivityFeedWidget extends Widget implements HasActions, HasForms, HasInfolists
+{
+    use HasTimelineRecords; // @todo: Refactor
+    use CanLoadTimelineRecords; // @todo: Refactor
```
[the remark this answers is not in the document]
Comment (danharrin) [reply app-modules/student-data-model/src/Filament/Resources/EducatableResource/Widgets/EducatableActivityFeedWidget.php:62]: This refers to the timeline refactor ticket

File: app-modules/student-data-model/src/Filament/Resources/EducatableResource/Widgets/EducatableActivityFeedWidget.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    use InteractsWithInfolists;
+
+    protected static string $view = 'student-data-model::filament.resources.educatable-resource.widgets.educatable-activity-feed-widget';
+
+    #[Locked]
+    public Educatable&Model $educatable;
+
+    public function mount(): void
+    {
+        // @todo: Refactor
```
[the remark this answers is not in the document]
Comment (danharrin) [reply app-modules/student-data-model/src/Filament/Resources/EducatableResource/Widgets/EducatableActivityFeedWidget.php:74]: As above

File: app-modules/student-data-model/src/Filament/Resources/EducatableResource/Widgets/EducatableActivityFeedWidget.php [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            Engagement::class,
+            EngagementResponse::class,
+            AlertHistory::class,
+            TaskHistory::class,
+            Interaction::class,
+        ];
+        // @endtodo
+    }
+
+    // @todo: Refactor
```
[the remark this answers is not in the document]
Comment (danharrin) [reply app-modules/student-data-model/src/Filament/Resources/EducatableResource/Widgets/EducatableActivityFeedWidget.php:86]: As above

**Review by Orrison — CHANGES_REQUESTED** [review]
Some strange UI issues on mobile:
Screenshot 2024-11-26 at 10 40 04 AM

Also there is a console error when opening the timeline full view:
[test.advisingapp.local-1732635738177.log](https://github.com/user-attachments/files/17921988/test.advisingapp.local-1732635738177.log)

**Comment by danharrin** [comment]
I have addressed both @Orrison

**Review by Orrison — APPROVED** [review]
Great work on this one @danharrin!

**Pull request merged**